extensions/kafka-client/runtime/src/main/java/io/quarkus/kafka/client/runtime/KafkaBindingConverter.java

```diff
@@ -0,0 +1,50 @@
+package io.quarkus.kafka.client.runtime;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import io.quarkus.kubernetes.service.binding.runtime.ServiceBinding;
+import io.quarkus.kubernetes.service.binding.runtime.ServiceBindingConfigSource;
+import io.quarkus.kubernetes.service.binding.runtime.ServiceBindingConverter;
+
+public class KafkaBindingConverter implements ServiceBindingConverter {
+
+    @Override
+    public Optional<ServiceBindingConfigSource> convert(List<ServiceBinding> serviceBindings) {
+        Optional<ServiceBinding> matchingByType = ServiceBinding.singleMatchingByType("kafka", serviceBindings);
+        if (!matchingByType.isPresent()) {
+            return Optional.empty();
+        }
+
+        Map<String, String> properties = new HashMap<>();
+        ServiceBinding binding = matchingByType.get();
+
+        String bootstrapServers = binding.getProperties().get("bootstrapServers");
+        if (bootstrapServers == null) {
+            bootstrapServers = binding.getProperties().get("bootstrap-servers");
+        }
+        if (bootstrapServers != null) {
+            properties.put("kafka.bootstrap.servers", bootstrapServers);
+        }
+
+        String securityProtocol = binding.getProperties().get("securityprotocol");
```

Comment:
why is it boostrapServers but securityprotocol?

Reply:
I just used what was mentioned in https://github.com/quarkusio/quarkus/issues/15631#issuecomment-799305457